deb/debian/preinst

```diff
@@ -0,0 +1,53 @@
+#!/bin/sh
+# preinst script for stackdriver-agent
+#
+# see: dh_installdeb(1)
+
+set -e
+
+# summary of how this script can be called:
+#        * <preinst> `install'
+#        * <preinst> `install' <old-version> <new-version>
+#        * <new-preinst> `upgrade' <old-version> <new-version>
+#        * <old-preinst> `abort-upgrade' <old-version>
+# for details, see http://www.debian.org/doc/debian-policy/ or
+# the debian-policy package
+
+. /usr/share/debconf/confmodule
+
+OPT_COLLECTD="/opt/stackdriver/collectd/etc"
+ARCHIVE_DIR="/etc/stackdriver/archive"
+
+case "$1" in
+    install)
+        ;;
+
+    upgrade)
```

Comment:
Is it worth adding logging (e.g., via "echo ...") so that users of these commands know wht we're doing? In general, I'd say this wouldn't be necessary, but since we're making a change to the way things used to work, maybe it's worth it in this case?

Reply:
So users will never invoke these commands directly. Debian relies on these scripts to implement an API which it uses in its upgrade processes. Adding echos will littler user logs and show them things that they were never previously privy to. Should we still add the echos? WDYT?

Reply:
I would say that the only entity that would find those echoes useful is us, for debugging the flow, and we can do that in our local packages. So I wouldn't bother adding them in the master branch.

Reply:
SGTM.